### DISCUSSION TITLE
How does Milvus GPU work? Does it require to store all indexes in GPU RAM so the main limit is GPU memory?

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [milvus-io/milvus#36328] [Q&A and General discussion] [answered]
How does Milvus GPU work? Does it require to store all indexes in GPU RAM so the main limit is GPU memory? 
How much faster is it that regular GPU index?

**Comment by yhmo** [comment, marked as answer]
Yes, milvus loads GPU index into GPU RAM. If the GPU RAM is insufficient, you will get an error about load failure.

When you call search(), the request's vector is copied into GPU RAM to compute, then result is copied from GPU RAM to CPU RAM, and returned to the client. So, there is an extra cost for copying data between GPU RAM and CPU RAM.
GPU is faster than CPU for large number of parallel tasks. So, GPU index is suitable for large NQ search, NQ is the number of vectors you input to search().

For example, search 10000 vectors with one search request, GPU index is faster than CPU index:
```
vectors = [[random.random() for _ in range(dim)] for _ in range(10000)]
results = collection.search(
                data = vectors,
                anns_field = "xxx",
                param = search_params,
                limit = 10,
            )
```
